busmall.js

```diff
@@ -110,62 +108,66 @@ function displayTable() {
     ul.appendChild(li);
 
   }
-  section.appendChild(ul);
+  sectionTotal.setAttribute('style', 'border: 10px double black');
+  sectionTotal.appendChild(h3);
+  sectionTotal.appendChild(ul);
 }
 
-/* 
 function renderChart() {
   var ctx = document.getElementById('chart').getContext('2d');
   var chart = new Chart(ctx, {
     type: 'bar',
     data: {
-      labels: [array of image names (global variable 'images') to be displayed for each bar]
+      labels: images,
+      datasets : [{
+        label: 'Items',
+        data: Item.totalVotes,
+        backgroundColor: ['black', 'red', 'black', 'red', 'black', 'red', 'black', 'red', 'black', 'red','black', 'red', 'black', 'red', 'black', 'red', 'black', 'red', 'black', 'red']
+      }]
     },
-    datasets : {
-      label: title of chart,
-      data: number of votes per image from global array created to store selection values from each object,
-      backgroundColors: [array of hex values for each value in data array],
-      borderColor: [same as background],
-      borderWidth: ...
-    }
     options: {
-      scales {
+      responsive: false,
+      maintainAspectRation: true,
+      title: {
+        display: true,
+        text: 'Your Most Selected Items',
+        fontSize: 20
+      },
+      scales: {
         yAxes: [{
           ticks: {
-            beginsAtZero: true;
+            beginAtZero: true,
+            stepSize: 1,
           }
         }]
       }
     }
   });
-} 
-
-function updateSelections() {
-  iterate through array of objects and record selection data to be referenced in chart.datasets
 }
-*/
 
 // selects a set of three random images
 function setImages() {
-  // generate new set of random images and count the ones displayed
   var randomNumbers = [];
 
+  // generates random number that is unique in the current set, and also different from the last set
   for (var i = 0; i < elArray.length; i++) {
     var random = Math.floor(Math.random() * Item.allItems.length);
-    if (randomNumbers.includes(random)) {
+    if (randomNumbers.includes(random) || Item.lastDisplayed.includes(random)) {
       i -= 1;
     } else {
       randomNumbers.push(random);
-      //Item.allItems[random].lastDisplayed.push(random);
-      // if ... clear lastDisplayed array
 
+      if (i === elArray.length - 1) {
+        for (var x = 0; x < randomNumbers.length; x++) {
+          Item.lastDisplayed[x] = randomNumbers[x];
+        }
+      }
+      console.log(Item.lastDisplayed);
       Item.allItems[random].displayed += 1;
       console.log('Displayed: ' + Item.allItems[random].alt + '- ' + Item.allItems[random].displayed);
     }
   }
 
-  // put this in for loop
-
   imgEl1.setAttribute('src', Item.allItems[randomNumbers[0]].src);
   imgEl1.alt = Item.allItems[randomNumbers[0]].alt;
```

Comment:
You can move these lines of code inside the `else` statement inside your `for` loop instead of repeating them here.